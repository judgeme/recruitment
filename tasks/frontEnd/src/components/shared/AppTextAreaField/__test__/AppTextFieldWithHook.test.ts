import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import { defineComponent } from 'vue';
import { useForm } from 'vee-validate';
import * as yup from 'yup';
import AppTextFieldWithHook from '@/components/shared/AppTextField/WithHook.vue';

describe('AppTextFieldWithHook.vue', () => {
  // Create a validation schema
  const schema = yup.object({
    email: yup.string().email('Invalid email').required('Email is required'),
  });
  // Create a simple form component using VeeValidate
  const TestForm = defineComponent({
    components: { AppTextFieldWithHook },
    setup() {
      const { handleSubmit } = useForm({ validationSchema: schema });
      const onSubmit = handleSubmit((values) => {
        console.log(values);
      });

      return { onSubmit };
    },
    template: `
      <form @submit="onSubmit">
        <app-text-field-with-hook name="email" label="Email" type="email" />
        <button type="submit">Submit</button>
      </form>
    `,
  });

  it('renders AppTextField with props and handles validation', async () => {


    // Mount the form component
    const wrapper = mount(TestForm);

    // Check if the AppTextFieldWithHook component is rendered
    const appTextFieldWithHook = wrapper.findComponent(AppTextFieldWithHook);
    expect(appTextFieldWithHook.exists()).toBe(true);

     // Find the input element
    const input = appTextFieldWithHook.find('input');
    expect(input.exists()).toBe(true);

    // Trigger blur to test validation
    await input.setValue('invalid-email');
    await input.trigger('update:modelValue');
    

    // Check if the validation error message is displayed
    // const errorMessage = appTextFieldWithHook.find('text-xs');
    // expect(errorMessage.exists()).toBe(true);
    // expect(errorMessage.text()).toBe('Invalid email');

    // // Correct the email and blur again
    // await input.setValue('test@example.com');
    // await input.trigger('blur');

    // Check if the validation error message is removed
    // expect(errorMessage.exists()).toBe(false);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
