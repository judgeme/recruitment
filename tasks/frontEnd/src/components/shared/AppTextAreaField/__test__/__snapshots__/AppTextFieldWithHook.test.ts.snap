// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppTextFieldWithHook.vue > renders AppTextField with props and handles validation 1`] = `
"<form>
  <div class="mb-2.5 w-full">
    <div class="relative"><label class="block tracking-wide text-primary text-anchor font-medium mt-0 mb-1"><span class="text-ellipsis">Email</span>
        <!--v-if-->
      </label><input class="appearance-none block w-full py-[15px] px-5 font-inherit font-light text-input text-primary bg-white border outline-0 transition-colors focus:bg-fairy placeholder:text-grey-deep disabled:cursor-not-allowed border-grey-60" name="email" type="email" autocomplete="off"></div>
    <!--v-if-->
  </div><button type="submit">Submit</button>
</form>"
`;
